app/code/Magento/Catalog/Controller/Adminhtml/Product/Attribute/Save.php

```diff
@@ -124,9 +141,18 @@ public function execute()
                 }
             }
 
+            /* @var $model ModelAttribute */
+            $model = $this->attributeFactory->create();
+
             $attributeId = $this->getRequest()->getParam('attribute_id');
-            $attributeCode = $this->getRequest()->getParam('attribute_code')
-                ?: $this->generateCode($this->getRequest()->getParam('frontend_label')[0]);
+            if($attributeId){
+                $model->load($attributeId);
+                $attributeCode = $model->getAttributeCode();
+            }else{
+                $attributeCode = $this->getRequest()->getParam('attribute_code')
```

Comment:
Are we actually passing `attribute_code` in some scenarios in current implementation?

Reply:
After of this, in this function checks if the attributeCode is valid or not using Zend_Validate_Regex. I think that this I can improve it in the next commit.

Reply:
I mean, is `attribute_code` present in UI or it is only auto-generated and then displayed as disabled field?

Reply:
Ah, ok, just checked admin may enter attribute code manually when creating new attribute.